temporalio/contrib/openai_agents/_temporal_model_stub.py

```diff
@@ -65,23 +65,30 @@ async def get_response(
         prompt: Optional[ResponsePromptParam],
     ) -> ModelResponse:
         def get_summary(
-            input: Union[str, list[TResponseInputItem], dict[str, str]],
+            input: Union[str, list[TResponseInputItem]],
         ) -> str:
             ### Activity summary shown in the UI
             try:
                 max_size = 100
                 if isinstance(input, str):
                     return input[:max_size]
                 elif isinstance(input, list):
-                    seq_input = cast(Sequence[Any], input)
-                    last_item = seq_input[-1]
-                    if isinstance(last_item, dict):
-                        return last_item.get("content", "")[:max_size]
-                    elif hasattr(last_item, "content"):
-                        return str(getattr(last_item, "content"))[:max_size]
-                    return str(last_item)[:max_size]
-                elif isinstance(input, dict):
-                    return input.get("content", "")[:max_size]
+                    content: Any = [
+                        item
+                        for item in input
+                        if (item.get("type") or "message") == "message"
+                    ][-1]
+                    if isinstance(content, dict):
+                        content = content.get("content", "")
+                    elif hasattr(content, "content"):
+                        content = getattr(content, "content")
+
+                    if isinstance(content, list):
+                        content = content[-1]
+
+                    if isinstance(content, dict) and content.get("text") is not None:
+                        content = content.get("text")
+                    return str(content)[:max_size]
```

Comment:
This code's pretty crazy and hard to see what it's supposed to be doing. Not blocking, but I think it would be a good candidate for a helper function with a unit test.